src/olympia/devhub/forms.py

```diff
@@ -586,10 +586,11 @@ def clean(self):
             self._clean_upload()
             xpi = parse_addon(self.cleaned_data['upload'], self.addon)
             # Make sure we don't already have the same non-rejected version.
-            if self.addon.versions.filter(version=xpi['version']).exclude(
-                    files__status=amo.STATUS_DISABLED):
+            if Version.unfiltered.filter(addon=self.addon,
+                                         version=xpi['version']):
```

Comment:
Note on performance, please use ``qset.exists()`` to avoid us fetching an unpredictable amount of objects from the database since Django will do something like ``len(list(qset))`` and evaluate the result.

Regarding the style, I'm a fan of avoiding line-breaks in expressions, e.g in ``if``, ``for`` statements and others and would do something like:

```python
version_exists = Version.unfiltered.filter(
    addon=self.addon, version=xpi['version']).exists()
if version_exists:
    ...
```
(or don't indend, depending on line-length)